ci/container/hang_webserver.py

```diff
@@ -20,6 +24,40 @@ def do_POST(self):
             self.send_response(200, 'OK')
             self.send_header('Content-Type', 'text/plain')
             self.end_headers()
+        elif self.path.startswith('/503'):
+            self.send_response(503)
+            self.send_header('Content-Type', 'text/plain')
+            self.end_headers()
+        elif self.path.startswith('/xml'):
+            self.send_response(200, '<error/>')
+            self.send_header('Content-Type', 'application/xml')
+            self.end_headers()
+        elif self.path.startswith('/resetCounter'):
+            HTTPRequestHandler.counter = 0
+            self.send_response(200, 'OK')
+            self.send_header('Content-Type', 'text/plain')
+            self.end_headers()
+        elif self.path.startswith('/eachThirdReturns200Others503'):
+            HTTPRequestHandler.counter += 1
+            if HTTPRequestHandler.counter % 3 == 0:
+                self.send_response(200, 'OK')
+                self.send_header('Content-Type', 'text/plain')
+                self.end_headers()
+            else:
+                self.send_response(503)
+                self.send_header('Content-Type', 'text/plain')
+                self.end_headers()
+        elif self.path.startswith('/eachThirdReturns200OthersHang'):
```

Comment:
Could you please give details about  /eachThirdReturns200OthersHang?

Reply:
I added description why I need this endpoint